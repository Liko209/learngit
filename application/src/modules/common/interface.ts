/*
 * @Author: Andy Hu(andy.hu@ringcentral.com)
 * @Date: 2019-03-04 16:12:23
 * Copyright Ã‚Â© RingCentral. All rights reserved.
 */
const CLIENT_SERVICE = 'window_service';
interface IClientService {
  focus: () => void;
  invokeApp: (url: string, options: InvokeAppOpts) => void;
}
interface InvokeAppOpts {
  fallback: Function;
}
export { CLIENT_SERVICE, IClientService, InvokeAppOpts };
