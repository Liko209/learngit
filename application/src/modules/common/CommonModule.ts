/*
 * @Author: Andy Hu(andy.hu@ringcentral.com)
 * @Date: 2019-03-04 16:12:23
 * Copyright Ã‚Â© RingCentral. All rights reserved.
 */
import { AbstractModule } from 'framework/AbstractModule';

class CommonModule extends AbstractModule {
  bootstrap() {}
}
export { CommonModule };
