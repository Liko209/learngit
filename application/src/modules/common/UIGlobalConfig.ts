/*
 * @Author: ken.li
 * @Date: 2019-05-29 21:50:21
 * Copyright Â© RingCentral. All rights reserved.
 */

import { GlobalConfig } from 'sdk/module/config';

class UIGlobalConfig extends GlobalConfig {
  static subModuleName = '';
  static moduleName = 'UI';
}

export { UIGlobalConfig };
