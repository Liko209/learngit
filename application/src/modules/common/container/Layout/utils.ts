/*
 * @Author: Nello Huang (nello.huang@ringcentral.com)
 * @Date: 2019-08-22 17:26:59
 * Copyright Â© RingCentral. All rights reserved.
 */

import { SectionTabs, LeftNav } from './types';

function isSectionTabs(config: LeftNav): config is SectionTabs {
  return !!(config as SectionTabs).sections;
}

export { isSectionTabs };
