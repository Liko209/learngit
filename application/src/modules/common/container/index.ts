/*
 * @Author: Valor Lin (valor.lin@ringcentral.com)
 * @Date: 2019-01-14 19:54:01
 * Copyright Â© RingCentral. All rights reserved.
 */
export * from './DocumentTitle';
export * from './ErrorPage';
