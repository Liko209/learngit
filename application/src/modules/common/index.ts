/*
 * @Author: Valor Lin (valor.lin@ringcentral.com)
 * @Date: 2019-01-14 19:36:49
 * Copyright Â© RingCentral. All rights reserved.
 */
export * from './container';
export * from './service';
export * from './module.config';
