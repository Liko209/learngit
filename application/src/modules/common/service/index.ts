/*
 * @Author: Andy Hu(andy.hu@ringcentral.com)
 * @Date: 2019-03-04 16:12:23
 * Copyright Ã‚Â© RingCentral. All rights reserved.
 */
export * from './ClientService';
