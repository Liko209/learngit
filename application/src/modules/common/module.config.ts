/*
 * @Author: Andy Hu(andy.hu@ringcentral.com)
 * @Date: 2019-03-04 16:12:23
 * Copyright Â© RingCentral. All rights reserved.
 */

import { CommonModule } from './CommonModule';
import { ClientService } from './service';
import { CLIENT_SERVICE } from './interface';

const config = {
  entry: CommonModule,
  provides: [{ name: CLIENT_SERVICE, value: ClientService }],
};

export { config };
