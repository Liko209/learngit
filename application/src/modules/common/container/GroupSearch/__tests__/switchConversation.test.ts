/*
 * @Author: Wayne Zhou (wayne.zhou@ringcentral.com)
 * @Date: 2019-08-25 23:23:15
 * Copyright Â© RingCentral. All rights reserved.
 */

import { testable, test } from 'shield';
import { Dialog } from '@/containers/Dialog';
import {
  switchConversationHandler,
  switchToConversation,
  getDefaultList,
} from '../switchConversationHandler';
import portalManager from '@/common/PortalManager';
import { goToConversationWithLoading } from '@/common/goToConversation';
import { isDialogOpen } from '@/containers/Dialog/utils';
import { mockService } from 'shield/sdk';
import { ServiceConfig } from 'sdk/module/serviceLoader';
import prettyFormat from 'pretty-format';

jest.mock('@/containers/Dialog/utils');
jest.mock('@/common/goToConversation');

describe('switch conversation handler', () => {
  const searchService = {
    name: ServiceConfig.SEARCH_SERVICE,
    doFuzzySearchAllGroups() {},
  };
  @testable
  class openGroupSearch {
    @test('should open group search when called')
    t1() {
      jest.spyOn(Dialog, 'simple').mockReturnValue({ dismiss: () => {} });
      switchConversationHandler();
      expect(Dialog.simple).toHaveBeenCalled();
    }

    @test('should not open GroupSearch when it has been opened')
    t2() {
      jest.spyOn(Dialog, 'simple').mockReturnValue({ dismiss: () => {} });
      jest.spyOn(portalManager, 'isOpened').mockReturnValue(true);
      (isDialogOpen as jest.Mock) = jest.fn(() => true);
      switchConversationHandler();
      expect(Dialog.simple).not.toHaveBeenCalled();
    }

    @test(
      'should close all dialog and call gotoConversationWithLoading when call switchToConversation()',
    )
    async t3() {
      const conversationId = 123;
      jest.spyOn(portalManager, 'dismissAll').mockImplementation(jest.fn());
      switchToConversation({ id: conversationId });
      expect(portalManager.dismissAll).toHaveBeenCalled();
      setTimeout(() => {
        expect(goToConversationWithLoading).toHaveBeenCalled();
      });
    }

    @(test.each`
      isThereDialogOpen | isMeOpen | expected
      ${true}           | ${true}  | ${false}
      ${true}           | ${false} | ${true}
      ${false}          | ${true}  | ${false}
      ${false}          | ${false} | ${false}
    `('should have correct return value when called'))
    t4({ isThereDialogOpen, isMeOpen, expected }) {
      jest.spyOn(Dialog, 'simple').mockReturnValue({ dismiss: () => {} });
      jest.spyOn(portalManager, 'isOpened').mockReturnValue(isMeOpen);
      (isDialogOpen as jest.Mock) = jest.fn(() => isThereDialogOpen);
      expect(switchConversationHandler()).toBe(expected);
    }

    @test(
      'should call searchService.doFuzzySearchAllGroups correctly when called',
    )
    @mockService(searchService, 'doFuzzySearchAllGroups')
    async getDefaultListTest() {
      searchService.doFuzzySearchAllGroups = jest.fn(() => []);
      await getDefaultList();
      expect(
        prettyFormat(searchService.doFuzzySearchAllGroups.mock.calls[0]),
      ).toMatchSnapshot();
    }
  }
});
