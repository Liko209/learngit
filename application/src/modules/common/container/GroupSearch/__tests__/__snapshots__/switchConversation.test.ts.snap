// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch conversation handler openGroupSearch should call searchService.doFuzzySearchAllGroups correctly when called 1`] = `
"Array [
  undefined,
  Object {
    \\"fetchAllIfSearchKeyEmpty\\": true,
    \\"myGroupsOnly\\": true,
    \\"recentFirst\\": false,
    \\"sortFunc\\": [Function recentFirstSorter],
  },
]"
`;
